frontend/packages/policy-editor/src/components/PolicyCardRules/PolicyRule/PolicyAccessPackages/policyAccessPackageUtils.ts

```diff
@@ -0,0 +1,33 @@
+import type { PolicyAccessPackageArea, PolicyAccessPackageAreaGroup } from '@altinn/policy-editor';
+
+const isStringMatch = (matchString: string, searchString: string): boolean => {
+  return matchString.toLowerCase().includes(searchString.toLowerCase());
+};
+
+export const filterAccessPackagesBySearchString = (
+  accessPackageAreas: PolicyAccessPackageArea[],
+  searchString: string,
+): PolicyAccessPackageArea[] => {
+  return accessPackageAreas.reduce(
+    (areas: PolicyAccessPackageArea[], area): PolicyAccessPackageArea[] => {
+      const matchingPackages = area.packages.filter(
+        (pack) =>
+          !searchString ||
+          isStringMatch(pack.name, searchString) ||
+          isStringMatch(pack.description, searchString),
+      );
+      const returnAreas = [...areas];
+      if (matchingPackages.length > 0) {
+        returnAreas.push({ ...area, packages: matchingPackages });
+      }
+      return returnAreas;
+    },
+    [],
+  );
+};
```

Comment:
This function does more than one thing and therefore it should be split into more functions 😄